cmd/opentelemetry/app/exporter/kafkaexporter/kafka_exporter_test.go

```diff
@@ -51,38 +65,174 @@ func TestDefaultConfig(t *testing.T) {
+				QueueSettings:    queueSettings,
+				RetrySettings:    exporterhelper.CreateDefaultRetrySettings(),
+				Authentication: otelKafkaExporter.Authentication{
+					PlainText: &otelKafkaExporter.PlainTextConfig{
+						Username: "user",
+						Password: "from-jaeger-flag",
+					},
+					Kerberos: &otelKafkaExporter.KerberosConfig{
+						ServiceName: "",
+						Username:    "",
+						Realm:       "jaeger",
+						ConfigPath:  "/etc/foo",
+						KeyTabPath:  "",
+					},
+					TLS: &configtls.TLSClientSetting{
+						Insecure: true,
+						TLSSetting: configtls.TLSSetting{
+							CAFile:   "ca.crt",
+							KeyFile:  "key.crt",
+							CertFile: "",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			v, c := jConfig.Viperize(AddFlags, flags.AddConfigFileFlag)
+			err := c.ParseFlags(tt.jFlags)
+			require.NoError(t, err)
 
+			err = flags.TryLoadConfigFile(v)
+			require.NoError(t, err)
+
+			factory := &Factory{
+				Wrapped: otelKafkaExporter.NewFactory(),
+				Viper:   v,
+			}
+
+			factories, err := componenttest.ExampleComponents()
+			factories.Exporters[TypeStr] = factory
+			cfg, err := configtest.LoadConfigFile(t, tt.oConfigFile, factories)
+			require.NoError(t, err)
+			require.NotNil(t, cfg)
+
+			kafkaCfg := cfg.Exporters[TypeStr].(*otelKafkaExporter.Config)
+			require.Equal(t, tt.expected, kafkaCfg)
+		})
+	}
+}
+
+func TestFactoryPassthrough(t *testing.T) {
 	factory := &Factory{
-		Wrapped: otelKafkaExporter.NewFactory(),
-		Viper:   v,
+		Wrapped: &componenttest.ExampleExporterFactory{},
 	}
 
-	factories, err := componenttest.ExampleComponents()
-	factories.Exporters[TypeStr] = factory
-	cfg, err := configtest.LoadConfigFile(t, path.Join(".", "testdata", "config.yaml"), factories)
-	require.NoError(t, err)
-	require.NotNil(t, cfg)
-
-	kafkaCfg := cfg.Exporters[TypeStr].(*otelKafkaExporter.Config)
-	require.NotNil(t, kafkaCfg)
-
-	assert.Equal(t, TypeStr, kafkaCfg.Name())
-	assert.Equal(t, "jaeger-prod", kafkaCfg.Topic)
-	assert.Equal(t, "emojis", kafkaCfg.Encoding)
-	assert.Equal(t, []string{"foo", "bar"}, kafkaCfg.Brokers)
-	assert.Equal(t, "user", kafkaCfg.Authentication.PlainText.Username)
-	assert.Equal(t, "123", kafkaCfg.Authentication.PlainText.Password)
-	assert.Equal(t, "ca.crt", kafkaCfg.Authentication.TLS.CAFile)
-	assert.Equal(t, "key.crt", kafkaCfg.Authentication.TLS.KeyFile)
-	assert.Equal(t, "cert.crt", kafkaCfg.Authentication.TLS.CertFile)
-	assert.Equal(t, true, kafkaCfg.Authentication.TLS.Insecure)
-	assert.Equal(t, "jaeger", kafkaCfg.Authentication.Kerberos.Realm)
-	assert.Equal(t, "/etc/foo", kafkaCfg.Authentication.Kerberos.ConfigPath)
-	assert.Equal(t, "from-jaeger-config", kafkaCfg.Authentication.Kerberos.Username)
+	actualLogs, _ := factory.CreateLogsExporter(context.Background(), component.ExporterCreateParams{}, nil)
+	expectedLogs, _ := factory.Wrapped.CreateLogsExporter(context.Background(), component.ExporterCreateParams{}, nil)
```

Comment:
Should there be an `assert.NoError(t, err)`?

That is, could `err != nil` while `actualLogs == expectedLogs` (e.g. both `nil`)?

Reply:
No, it's not really testing that.  It's really just confirming that the `ExporterFactory` is just a thin wrapper around the internal `ExporterFactory`.

But you do bring up a good point.  I should be testing both the errs and the return.